arelle/plugin/validate/ESEF_2022/Util.py

```diff
@@ -55,41 +60,98 @@ def isInEsefTaxonomy(val: ValidateXbrl, modelObject: ModelObject | None) -> bool
+
+def validateImage(baseUrl:Optional[str], image: str, modelXbrl: ModelXbrl, val:ValidateXbrl, elt:_Element, evaluatedMsg:str, contentOtherThanXHTMLGuidance:str) -> None:
+    """
+    image: either an url or base64 in data:image style
+    """
+    minExternalRessourceSize = val.authParam["minExternalResourceSizekB"]
+    if minExternalRessourceSize != -1:
+        # transform kb to b
+        minExternalRessourceSize = minExternalRessourceSize * 1024
+    if scheme(image) in ("http", "https", "ftp"):
+        modelXbrl.error("ESEF.4.1.6.xHTMLDocumentContainsExternalReferences" if val.unconsolidated
+                        else "ESEF.3.5.1.inlineXbrlDocumentContainsExternalReferences",
+                        _("Inline XBRL instance documents MUST NOT contain any reference pointing to resources outside the reporting package: %(element)s"),
+                        modelObject=elt, element=elt.tag, evaluatedMsg=evaluatedMsg,
+                        messageCodes=("ESEF.3.5.1.inlineXbrlDocumentContainsExternalReferences",
+                                      "ESEF.4.1.6.xHTMLDocumentContainsExternalReferences"))
+    elif not image.startswith("data:image"):
+        # presume it to be an image file, check image contents
+        try:
+            base = baseUrl
+            normalizedUri = modelXbrl.modelManager.cntlr.webCache.normalizeUrl(image, base)
+            if not modelXbrl.fileSource.isInArchive(normalizedUri):
+                normalizedUri = modelXbrl.modelManager.cntlr.webCache.getfilename(normalizedUri)
+            imglen = 0
+            with modelXbrl.fileSource.file(normalizedUri, binary=True)[0] as fh:
+                imgContents = fh.read()
+                imglen += len(imgContents)
+                checkImageContents(normalizedUri, modelXbrl, elt, os.path.splitext(image)[1], True, imgContents,
+                                   val.consolidated, val)
+                imgContents = None  # deref, may be very large
+            if minExternalRessourceSize == -1 or imglen < minExternalRessourceSize:
+                modelXbrl.warning(
+                    "%s.imageIncludedAndNotEmbeddedAsBase64EncodedString" % contentOtherThanXHTMLGuidance,
+                    _("Images SHOULD be included in the XHTML document as a base64 encoded string unless their size exceeds the minimum size for the authority (%(maxImageSize)s): %(file)s."),
+                    modelObject=elt, maxImageSize=minExternalRessourceSize, file=os.path.basename(normalizedUri), evaluatedMsg=evaluatedMsg)
+        except IOError as err:
+            modelXbrl.error(f"{contentOtherThanXHTMLGuidance}.imageFileCannotBeLoaded",
+                            _("Error opening the file '%(src)s': %(error)s"),
+                            modelObject=elt, src=image, error=err, evaluatedMsg=evaluatedMsg)
+    else:
+        m = imgDataMediaBase64Pattern.match(image)
+        if not m or not m.group(2):
```

Comment:
nit: readability could be improved a little here by unpacking to named vars (or using named regex capture groups) and reusing them below:
```python
(imgMimeType, isBase64, imgData) = m.group(1, 2, 3) if m is not None else (None, None, None)
```

Reply:
The suggested change has been done!